File: 프로그래머스/unrated/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    my_string = list(my_string)
    for s,e in queries: 
        if s>0 :
            change = my_string[e:s-1:-1]
            idx = s 
            for c in change:
                my_string[idx] = c
                idx+=1
        else:
            change = my_string[e::-1]
            idx=s
            for c in change:
                my_string[idx] = c
                idx+=1
                
        
    return ''.join(my_string)

File: 프로그래머스/unrated/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
    let answer = [];
    for([s,e] of queries){
        for(let i=0;i<s;i++){
            answer.push(my_string[i]);
        }
        for(let i=e;i>=s;i--){
            answer.push(my_string[i]);
        }
        for(let i=e+1;i<my_string.length;i++){
            answer.push(my_string[i]);
        }
        my_string = answer.join('');
        answer = [];
    }
    return my_string;
}